testing/matrix-sdk-integration-testing/src/tests/sliding_sync/room.rs

```diff
@@ -538,3 +551,354 @@ async fn test_room_notification_count() -> Result<()> {
+        )
+        .build()
+        .await?;
+
+    let s = sliding_alice.clone();
+    spawn(async move {
+        let stream = s.sync();
+        pin_mut!(stream);
+        while let Some(up) = stream.next().await {
+            warn!("alice received update: {up:?}");
+        }
+    });
+
+    let s = sliding_bob.clone();
+    spawn(async move {
+        let stream = s.sync();
+        pin_mut!(stream);
+        while let Some(up) = stream.next().await {
+            warn!("bob received update: {up:?}");
+        }
+    });
+
+    // alice creates a room and invites bob.
+    let alice_room = alice
+        .create_room(assign!(CreateRoomRequest::new(), {
+            invite: vec![bob.user_id().unwrap().to_owned()],
+            is_direct: true,
+            preset: Some(RoomPreset::TrustedPrivateChat),
+        }))
+        .await?;
+    alice_room.enable_encryption().await.unwrap();
+
+    let (stream, entries) = alice_sync_service
+        .room_list_service()
+        .all_rooms()
+        .await
+        .unwrap()
+        .entries_with_dynamic_adapters(10, alice.roominfo_update_receiver());
+
+    entries.set_filter(new_filter_all());
+    pin_mut!(stream);
+
+    // Wait shortly so the manual roominfo update is triggered before we load the
+    // stream.
+    sleep(Duration::from_secs(1)).await;
+
+    // Stream only has the initial Reset entry.
+    assert_eq!(
+        timeout(Duration::from_millis(100), stream.next()).await,
+        Ok(Some(vec![VectorDiff::Reset {
+            values: vec![RoomListEntry::Filled(alice_room.room_id().to_owned())].into()
+        }]))
+    );
+    assert_pending!(stream);
+
+    sleep(Duration::from_secs(1)).await;
+    let alice_room = alice.get_room(alice_room.room_id()).unwrap();
+    let bob_room = bob.get_room(alice_room.room_id()).unwrap();
+    bob_room.join().await.unwrap();
+
+    sleep(Duration::from_secs(1)).await;
+    assert_eq!(alice_room.state(), RoomState::Joined);
+    assert!(alice_room.is_encrypted().await.unwrap());
+    assert_eq!(bob_room.state(), RoomState::Joined);
+    // Room update for join
+    assert_eq!(
+        timeout(Duration::from_millis(100), stream.next()).await,
+        Ok(Some(vec![VectorDiff::Set {
+            index: 0,
+            value: RoomListEntry::Filled(alice_room.room_id().to_owned())
+        }]))
+    );
+    assert_pending!(stream);
+
+    // Send a message, it should arrive
+    let event = bob_room.send(RoomMessageEventContent::text_plain("hello world")).await?;
+
+    sleep(Duration::from_secs(1)).await;
+
+    // Latest event is set now
+    assert_eq!(alice_room.latest_event().unwrap().event_id(), Some(event.event_id));
+
+    // Stream has the room again, but no second event
+    // TODO: Synapse sometimes sends the same event two times. This is the
+    // workaround:
+    assert!(timeout(Duration::from_millis(100), stream.next()).await.unwrap().unwrap().len() > 0);
```

Comment:
Could you use temporary variables here? It's not clear what's the final thing we're checking against...